src/document/DocumentCommandHandlers.js

```diff
@@ -1596,6 +1600,34 @@ define(function (require, exports, module) {
         ProjectManager.showInTree(MainViewManager.getCurrentlyViewedFile(MainViewManager.ACTIVE_PANE));
     }
 
+    /** Download selected file or folder structure **/
+    function handleFileDownload() {
+        var selectedItem = ProjectManager.getSelectedItem();
+        var path = selectedItem._path;
+        try {
+            fs.stat(path, function(err, stats) {
+                if (stats.type === "DIRECTORY") {
+                    ArchiveUtils.archive(path);
+                } else {
+                    // Prepare file for download
+                    fs.readFile(path, {encoding: null}, function(err, data){
+                        if (err) {
+                            console.error(err);
```

Comment:
Same as above, you probably want to show a modal dialog in brackets that says there was a problem.